src/librustc/hir/map/collector.rs

```diff
@@ -27,6 +27,9 @@ pub struct NodeCollector<'ast> {
     pub map: Vec<MapEntry<'ast>>,
     /// The parent of this node
     pub parent_node: NodeId,
+    /// Whether to completely ignore nested items,
+    /// such as in decoded HIR fragments from metadata
+    pub ignore_nested_items: bool
```

Comment:
Can you elaborate just a touch on this comment? I always find these kind of booleans that switch between "two similar modes" kind of creepy, so having a good explanation of why you might pick one or the other is great. In particular, I'd change this from "such as in", which suggests that there are other times one might want it to be `true`. In fact, that's the only time -- right? So maybe something like:

"If true, completely ignore nested items. We do this when loading HIR from metadata since in that case we only want the HIR for one specific item (and not the ones nested inside of it)."

That said, I think that HIR encoded into metadata doesn't have nested items? Oh, I guess we stopped folding it now, huh?